[22기 최민재] 리뷰 반영, Conflict 해결 완료

rious275 opened the pull request:
## 수정 사항 간략한 한줄 요약

- 리뷰 및 컨플릭 해결 완료입니다.
		 
## 수정 사항들 자세한 내용

## 기타 질문 및 특이 사항

- 컨플릭 해결 관련해서, 내용이 수정 전과 수정 후가 같은 내용인데.. 왜 컨플릭 났는지 의문이네요.

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.
- [x] Git rebase와 squash를 했고 커밋 수가 하나 인것을 확인 했습니다.

saengmotmi commented:
@rious275 리뷰 요청하는 내용이면 라벨 달아주세요

saengmotmi commented on .eslintrc:
```diff
@@ -1,9 +1,12 @@
     "linebreak-style": 0,
     "prettier/prettier": ["error", { "endOfLine": "auto" }],
-    "react/react-in-jsx-scope": "off"
+    "react/react-in-jsx-scope": "off",
+    "react-hooks/rules-of-hooks": "error",
+    "react-hooks/exhaustive-deps": "warn"
```
린트 세팅 같은 경우 다른 팀원들에게도 공통되는 부분인데, 협의가 된 사항인가요?

rious275 replied:
이 부분 제가 테스트하다가 지우지 못했습니다 ㅜㅜ 제 쪽에서 삭제하겠습니다.

saengmotmi commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,184 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import styled from 'styled-components';
+import { theme, setFlex } from '../../styles/theme';
```
theme에서 작성한 내용들은 이렇게 import 하는게 아니라 ThemeProvider를 통해 각각의 스타일드 컴포넌트로 직접 주입된 props를 사용하는게 좋습니다
```js
const Foo = styled.div`
  ${({ theme }) => theme.setFlex()}
```

rious275 replied:
확인했습니다!

saengmotmi commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,184 @@
+`;
+
+const Article = styled.section`
+  display: flex;
+  flex-direction: column;
```
flex 사용하지 않고 자식 요소들의 block 속성을 이용하면 세로 배열 할 수 있을거 같습니다

rious275 replied:
알겠습니다!

saengmotmi commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,184 @@
+`;
+
+const Aside = styled.aside`
+  left: 700px;
```
- position 속성 없이 left 속성 작동하나요?
- 거의 모든 요소들에 가로, 세로 고정 폭이 주어져 있는데 가급적 부모 요소들에는 크기를 주지 말고, 자식요소 크기 + padding, margin 값으로 레이아웃 잡으셔야 합니다

rious275 replied:
넵 확인했습니다!!

saengmotmi commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,184 @@
+  ul {
+    margin: 24px 0;
+    display: flex;
+    flex-direction: row;
```
direction row는 flex 사용하면 기본 값입니다 삭제해주세요

rious275 replied:
넵!

saengmotmi commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,184 @@
+  color: #fff;
+`;
+
+const KakaoMap = styled.div`
+  width: 100%;
```
div 태그에 가로 100% 필요한지도 봐주세요...

rious275 replied:
이리 저리 적용하다보니 필요없는 style이 많네요.. ㅜㅜ 수정하겠습니다

saengmotmi reviewed (commented):
코멘트 확인해주세요 :)

yeonuk-hwang commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,201 @@
+  const KakaoMapContainer = useRef(null);
+
+  useEffect(() => {
+    const markerPosition = new window.kakao.maps.LatLng(37.5063, 127.05367);
+    const marker = new window.kakao.maps.Marker({
+      position: markerPosition,
+    });
```
```suggestion
    const marker = new window.kakao.maps.Marker({
      position: new window.kakao.maps.LatLng(37.5063, 127.05367);
    });
```
굳이 변수 선언해서 담을필요 없어보입니다!

yeonuk-hwang commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,201 @@
 };
 
+const Container = styled.div`
+  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
+  margin: 50px;
+  line-height: 1.2;
+  ${({ theme }) => theme.setFlex('center', 'none')}
+`;
```
css 속성 작성순서 컨벤션 맞춰주세요!
다른부분도 한번씩 보고 점검해주세요

rious275 replied:
확인하겠습니다!

yeonuk-hwang commented on src/pages/Detail/Modal/ModalShare.js:
```diff
@@ -0,0 +1,111 @@
+          setOpenModal(true);
+        }}
+        alt=""
+        src={process.env.PUBLIC_URL + 'images/icon_share.svg'}
```
```suggestion
        src='/images/icon_share.svg'
```
이렇게도 해도 적용될 것 같습니다! 혹시 PUBLIC_URL을 따로 기입하신 이유가 있나요?

rious275 replied:
1차 때 적용이 잘 안되었는데, 저걸 붙이니 잘 되더라고요. 
그런데 지금 코드 수정해보니까 저 부분 없이도 잘 구현 되네요...

yeonuk-hwang commented on src/pages/Detail/Modal/ModalShare.js:
```diff
@@ -0,0 +1,111 @@
+import styled from 'styled-components';
+
+const ModalShare = () => {
+  const [openModal, setOpenModal] = useState(false);
```
modal이 열렸는지 여부를 관리하는 boolean state의 이름이라면 isModalOpen등이 적합하지 않을까요?

rious275 replied:
직관적이네요.. 수정했습니다!

yeonuk-hwang commented on src/pages/Detail/Modal/ModalShare.js:
```diff
@@ -0,0 +1,111 @@
+
+const ModalShare = () => {
+  const [openModal, setOpenModal] = useState(false);
+  const [success, setSuccess] = useState(false);
```
무엇에 대한 success인지 알아볼 수 없는 이름 같습니다!

rious275 replied:
successTextCopy 로 변경했습니다!

yeonuk-hwang commented on src/pages/Detail/Modal/ModalShare.js:
```diff
@@ -0,0 +1,111 @@
+                readOnly
+              />
+              <button>복사</button>
+              {success === true && <Success>복사되었습니다</Success>}
```
success자체가 이미 boolean인데 ===으로 비교할 필요가 있을까요?
```suggestion
              {success && <Success>복사되었습니다</Success>}
```

rious275 replied:
수정했습니다!

yeonuk-hwang commented on src/pages/Detail/Modal/ModalShare.js:
```diff
@@ -0,0 +1,111 @@
+
+const ModalOverlay = styled.div`
+  position: absolute;
+  width: 100%;
```
block요소인데 width:100% 부여할 필요 없어보입니다.

rious275 replied:
이 부분은 width를 주지 않으면 화면에 올라오지 않더라구요. 어떤 문제일까요..?

yeonuk-hwang replied:
absolute 속성에 의한 동작으로 추측됩니다!
absolute 속성으로 인해 부모로부터 벗어났다고 판단이되어서 block이지만 width가 없는 것 처럼 인식이 되는 것 같습니다!
width:100%. 부여하시는게 맞아보이네요!

rious275 replied:
넵 감사합니다!

yeonuk-hwang reviewed (commented):
카카오맵 잘 활용해보셨네요!
리뷰 확인해주세요

yeonuk-hwang commented:
리뷰 요청하시는거면 라벨 붙여주세요!

yeonuk-hwang commented:
민재님 작업내용이 잘 안올라오는 것 같습니다!
작업하시는 내용 길어지시는건가요?
아니면 혹시 한 브랜치에서 너무 많은 작업을 하고 계실 수도 있습니다.
기능별로 브랜치 적절하게 분리하셔서 피드백 받으면서 작업하시는게 좋을거에요!

rious275 commented:
앗 넵 ㅜㅜ 처음부터 너무 한 브랜치에 했던 것 같아요.
한 번 올리고 다음 기능부터는 브랜치 분리하도록 하겠습니다!

On Mon, Jul 26, 2021 at 4:44 PM Yeonuk Hwang ***@***.***>
wrote:

> 민재님 작업내용이 잘 안올라오는 것 같습니다!
> 작업하시는 내용 길어지시는건가요?
> 아니면 혹시 한 브랜치에서 너무 많은 작업을 하고 계실 수도 있습니다.
> 기능별로 브랜치 적절하게 분리하셔서 피드백 받으면서 작업하시는게 좋을거에요!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/wecode-bootcamp-korea/22-2nd-Oneted-frontend/pull/6#issuecomment-886459198>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASOJM2ATZTSZEAISV6YKT5LTZUG5BANCNFSM5AVTNE5Q>
> .
>

yeonuk-hwang commented:
민재님 작업 중간이더라도 한번 푸시하시고 리뷰요청 하시는게 좋아보입니다!
피드백을 잘 못받고 계시는 것 같아서요!

yeonuk-hwang commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,246 @@
+import styled from 'styled-components';
+import ModalShare from './Modal/ModalShare';
+
+const options = {
```
변하지 않는 상수값이라면 대문자로 변수명을 설정해주는게 좋습니다.

yeonuk-hwang commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,246 @@
 
 const Detail = () => {
-  return <div></div>;
+  const [data, setData] = useState({});
```
data 라는 변수명은 좋지 않습니다.
모든 변수들은 다 데이터를 담고 있을건데 포괄적으로 data라고 하면 어떤 변수인지 알아볼 수가 없습니다.
해당 변수가 의미하는 걸 명확히 알아볼 수 있게 변수명을 지어주세요!

rious275 replied:
넵 처음 테스트 하면서 작업을 같이 진행하고 수정을 못했습니다.

jobData로 수정했습니다!

yeonuk-hwang commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,246 @@
 const Detail = () => {
-  return <div></div>;
+  const [data, setData] = useState({});
+  const [isModalOpen, setIsModalOpen] = useState(false);
```
이 변수명은 너무 좋습니다!
모달이 오픈됬는지 아닌지의 여부를 나타내는 boolean값이구나라고 변수명만 보고도 바로 알아볼 수 있네요!

yeonuk-hwang commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,246 @@
+  const KakaoMapContainer = useRef();
+
+  useEffect(() => {
+    fetch('data/data.json', {
```
파일명도 data말고 특정한 이름으로 설정해주는게 좋아보입니다

rious275 replied:
넵 이 부분은 테스트용 데이터인데, 파일명 수정은 진행 했습니다!

yeonuk-hwang commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,246 @@
+
+  useEffect(() => {
+    fetch('data/data.json', {
+      method: 'GET',
```
get은 기본값입니다. 그래서 따로 주입할 필요가 없습니다!

yeonuk-hwang commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,246 @@
+  }, []);
+
+  useEffect(() => {
+    const marker = new window.kakao.maps.Marker({
+      position: new window.kakao.maps.LatLng(37.5063, 127.05367),
+    });
+
+    marker.setMap(
+      new window.kakao.maps.Map(KakaoMapContainer.current, options)
+    );
+  }, []);
```
카카오맵 잘 쓰셨네요 👍

yeonuk-hwang commented on src/pages/Detail/Detail.js:
```diff
@@ -1,7 +1,246 @@
+            onClick={() => {
+              setIsModalOpen(true);
+            }}
+            alt=""
```
alt 값 채워주세요!

yeonuk-hwang replied:
채워주세요~

yeonuk-hwang commented on src/pages/Detail/Modal/ModalApply.js:
```diff
@@ -0,0 +1,208 @@
+    e.preventDefault();
+    const formData = new FormData();
+
+    for (let i = 0; i < addFile.length; i++) {
+      formData.append('file', addFile[i]);
+    }
```
```suggestion
    addFile.forEach(file => formData.append("file",file));
```
forEach 메소드를 이용해서도 할 수 있습니다!
기본적이고 절차적인 for 문보다는 forEach, map, reduce 등의 선언적인 메소드들을 잘 활용하시는게 좋습니다!

yeonuk-hwang commented on src/pages/Detail/Modal/ModalApply.js:
```diff
@@ -0,0 +1,208 @@
+
+    fetch('http://10.58.6.154:8000/resumes/file', {
+      method: 'POST',
+      body: formData,
```
파일전송 잘 되나요? 👍

yeonuk-hwang commented on src/pages/Detail/Modal/ModalApply.js:
```diff
@@ -0,0 +1,208 @@
+      body: formData,
+    })
+      .then(res => res.json())
+      .then(res => console.log('업로드 성공', res));
```
확인 끝난 콘솔은 지워주세요

yeonuk-hwang commented on src/pages/Detail/Modal/ModalApply.js:
```diff
@@ -0,0 +1,208 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+const ModalApply = props => {
```
```suggestion
const ModalApply = ({closeModal}) => {
```
매개변수 자리에서 바로 구조분해 할 수 있습니다.

yeonuk-hwang commented on src/pages/Detail/Modal/ModalApply.js:
```diff
@@ -0,0 +1,208 @@
+          </BodyTextContainer>
+          <div>
+            <SubmitTextBottom>첨부파일</SubmitTextBottom>
+            {/* <ul>
+              <li>내가 등록한 파일</li>
+              <li>이력서1</li>
+              <li>이력서2</li>
+              <li>이력서3</li>
+            </ul> */}
```
불필요한 주석 삭제해주세요

yeonuk-hwang commented on src/pages/Detail/Modal/ModalApply.js:
```diff
@@ -0,0 +1,208 @@
+          <form encType="multipart/form-data" onSubmit={onSubmit}>
+            <div>
+              {uploadValue &&
+                addFile.map(file => {
+                  return (
+                    <UploadList>
+                      <span>V</span>
+                      <div>{file.name}</div>
+                      <p>date</p>
+                      <p>첨부파일</p>
+                    </UploadList>
+                  );
+                })}
```
👍

rious275 replied:
거의 다 완료된 줄 알았는데 체크박스 상태 관리가 쉽지 않네요.. ㅎㅎ

yeonuk-hwang commented on src/pages/Detail/Modal/ModalApply.js:
```diff
@@ -0,0 +1,208 @@
+                    </UploadList>
+                  );
+                })}
+              <UploadBtn for="upload">파일업로드</UploadBtn>
+              <input
+                type="file"
+                id="upload"
+                multiple
+                onChange={onInputChange}
+                style={{ display: 'none' }}
+              />
```
- `id` 보다는 `className` 을 활용해주시기 바랍니다.
- HTML에서  `id` 는 문서 내에서 유일해야 합니다. 이러한 이유로 단일 요소를 다루는 경우 `id`를 사용할 수 있습니다.
- 하지만 정말로 해당 요소에 대한 이벤트나 스타일 핸들링이 프로젝트 전체에서 단 한번만 사용 되는지 충분히 고민해야합니다. 일반적으로 컴포넌트는 재사용 될 수 있기 때문에 같은 요소가 여러번 생성될 수 있습니다. 이러한 이유로 보통의 경우에는 `className` 을 사용합니다.
- label의 for를 위해서 사용하신거면
대상이되는 것을 라벨로 감싸면 for 속성필요없이 자동으로 인식됩니다.
`<UplodaBtn><input/></UploadBtn>`

rious275 replied:
오 .... 꿀팁 감사합니다

yeonuk-hwang commented on src/pages/Detail/Modal/ModalApply.js:
```diff
@@ -0,0 +1,208 @@
+
+export default ModalApply;
+
+const ModalApplyBg = styled.div``;
+
+const ModalOverlay = styled.div``;
```
스타일을 부여하지 않는 것들은 굳이 스타일드 컴포넌트로 만들어서 사용할 필요는 없습니다.

rious275 replied:
혹시 태그 용도를 알리기 위해 사용하는건 무의미할까요?

yeonuk-hwang replied:
그런용도로 사용할 수도 있겠지만, 일반적으로 태그에 스타일이 안부여된 태그는 단순히 감싸기 위한 용도의 div일 것 같아서 딱히 그렇게 할 상황이 많지는 않다고 생각됩니다!

yeonuk-hwang reviewed (commented):
리뷰 확인 및 수정해주세요!

younuk23 commented:
컨플릭트 해결해주세요~

yeonuk-hwang reviewed (commented):
컨플릭트 해결해주세요~

Commit:
- [Add] 채용공고 상세페이지 1차 레이아웃 작업 및 카카오 지도 API 적용 완료
- [Add] 공유하기(클립보드 복사) 기능 추가, 1차 리팩토링 진행
- [Add] 리팩토링 및 공유하기 기능 구현

- 구조 변경 전 저장
- test branch 파기전 저장

- [Add] PR 피드백 반영

- [Add] 파일 업로드 백엔드 통신 테스트 완료
- [Add] 파일 업로드 구현 완료 및 스타일 작업중

- [Refactor] 리뷰 반영, Conflict 해결

- 리뷰 반영 및 Conflict 해결

yeonuk-hwang commented:
머지 축하드립니다~
